draft-huitema-dprive-dnsoquic.md

```diff
@@ -351,6 +325,28 @@ There is no guarantee that the peer will receive the Stop Sending frame before c
 If it does, it SHOULD send a QUIC Stream Reset on the transaction specific stream, using the 
 error code DOQ_TRANSACTION_CANCELLED.
 
+## DoQ Error Codes
+
+The following error codes are defined for use when abruptly terminating streams,
+aborting reading of streams, or immediately closing connections:
+
+DOQ_NO_ERROR (0x00):
+: No error.  This is used when the connection or stream needs to be closed, but
+  there is no error to signal.
+
+DOQ_INTERNAL_ERROR (0x01):
+: The DoQ implementation encountered an internal error and is incapable of
+  pursuing the transaction or the connection.
+
+DOQ_TRANSACTION_CANCELLED (0x02):
+: Used in a Stop Sending request to signal that the originator of the query is
+  not anymore interested by the result. Also used by the recipient of the
+  request when issuing a Stream Reset in response to a Stop Sending request.
+
```

Comment:
I am having second thoughts about that one. There is no mechanism to cancel a transaction in DNS over TCP. Do we really need to invent one for DoQ? I think it would be better to only use the QUIC mechanisms for fatal errors, and adopt the general rule that all application errors are conveyed by DNS messages.